icu4j/main/classes/core/src/com/ibm/icu/number/NumberSkeletonImpl.java

```diff
@@ -947,6 +947,10 @@ private static void generateSkeleton(MacroProps macros, StringBuilder sb) {
             throw new UnsupportedOperationException(
                     "Cannot generate number skeleton with custom padder");
         }
+        if (macros.unitDisplayCase != null) {
```

Comment:
The code in this PR is treating an empty string as though it's a case. Thus when the unit test sets case to the empty string, the skeleton generator complains that it doesn't know how to generate skeletons for unitDisplayCase, rather than taking it as having no case set.

(This is unrelated to this PR, except for understanding the difference between unitDisplayCase() and usage(), since I compared to help me understand what's happening: I think I may have found a bug in "usage" handling. In C++, macros.usage.isSet() is used, where an empty string is treated as "not set". In Java, it might be treated as given. It looks like I might have failed to add a unit test case in C++ for `.usage("")` clearing the usage. In Java this test case is also missing. Thus in C++ it's treated as "unset usage", whereas in Java it's probably falling back to behaving like `.usage("default")`. Do you want to fix this or shall I?)

Reply:
We've found the solution.